pkg/gardenlet/controller/shoot/shoot/reconciler_delete.go

```diff
@@ -328,13 +328,13 @@ func (r *Reconciler) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 		deployKubeControllerManager = g.Add(flow.Task{
 			Name:         "Deploying Kubernetes controller manager",
 			Fn:           flow.TaskFn(botanist.DeployKubeControllerManager).RetryUntilTimeout(defaultInterval, defaultTimeout),
-			SkipIf:       !cleanupShootResources || !kubeControllerManagerDeploymentFound,
+			SkipIf:       !cleanupShootResources || (!kubeControllerManagerDeploymentFound && !controlPlaneDeploymentNeeded),
 			Dependencies: flow.NewTaskIDs(initializeSecretsManagement, deployCloudProviderSecret, waitUntilControlPlaneReady, initializeShootClients),
 		})
 		_ = g.Add(flow.Task{
 			Name:         "Scaling up Kubernetes controller manager",
 			Fn:           botanist.ScaleKubeControllerManagerToOne,
-			SkipIf:       !cleanupShootResources || !kubeControllerManagerDeploymentFound,
+			SkipIf:       !cleanupShootResources || (!kubeControllerManagerDeploymentFound && !controlPlaneDeploymentNeeded),
```

Comment:
```suggestion
			SkipIf:       !cleanupShootResources,
```

Do you think this will be sufficient?

Reply:
I tested it and it works also for previously stuck shoots.

Does it make sense to still keep the `kubeControllerManagerDeploymentFound` bool in
https://github.com/gardener/gardener/blob/f92e17b046bcb9460da8161120cab90988b52f75/pkg/gardenlet/controller/shoot/shoot/reconciler_delete.go#L370-L375
since this bool also deals with the KCM deployment in deletion.

Reply:
Sounds good to me👍